File: java/src/main/java/com/myriad/wdl/model/errors/WdlSyntaxError.java
package com.myriad.wdl.model.errors;

import org.antlr.v4.runtime.RecognitionException;

/** Syntax diagnostic produced while lexing or parsing WDL source. */
public class WdlSyntaxError extends WdlError {
  private final RecognitionException cause;

  /** Creates a syntax diagnostic and preserves the parser exception when available. */
  public WdlSyntaxError(
      String message, int line, int charPositionInLine, RecognitionException cause) {
    super(message, line, charPositionInLine);
    this.cause = cause;
  }

  /** Returns the underlying ANTLR recognition exception, if any. */
  public RecognitionException getCause() {
    return cause;
  }
}

File: python/src/wdl_model/model/errors/wdl_syntax_error.py
"""Syntax diagnostic produced while lexing or parsing WDL source."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Optional

from .wdl_error import WdlError


@dataclass
class WdlSyntaxError(WdlError):
    """Syntax diagnostic that preserves the underlying parser exception when available."""

    cause: Optional[Any] = None

    def getCause(self) -> Optional[Any]:
        """Return the underlying parser exception, if any."""
        return self.cause
